### PR TITLE
COMPUTE-471 fix dx run instance type by executable

### PR DESCRIPTION
**Pull request by YuxinShi0423**
Fix https://github.com/dnanexus/dx-toolkit/pull/1126
So `--instance-type-by-exectable` takes the proper format double mapping
```
$ dx run applet-GgKb14Q0k955yZgG779Gybz1 --instance-type mem2_hdd2_x2 --instance-type-by-executable '{"applet-GgKb14Q0k955yZgG779Gybz1": {"some_ep": "mem1_ssd1_v2_x2", "some_other_ep": "mem2_hdd2_x4"}}' --brief -y

job-GgKb1980k956yk8p5XZ7KJgB
```

### CODE REVIEW

File: src/python/dxpy/scripts/dx.py [hunk 1]
```diff
@@ -3229,10 +3229,26 @@ def run_body(args, executable, dest_proj, dest_path, preset_inputs=None, input_n
     # into the runtime systemRequirements
     requested_system_requirements = (requested_instance_type + requested_cluster_spec + requested_fpga_driver).as_dict()
 
-    # store runtime --instance-type-by-executable {executable:{entrypoint:{'instanceType':xxx}}} as systemRequirementsByExecutable 
+    if (args.instance_type and (args.instance_type_by_executable or cloned_system_requirements_by_executable)):
+        warning = BOLD("WARNING") + ": --instance-type argument {} may get overridden by\n".format(args.instance_type)
```
Comment (emiloslavsky) [inline src/python/dxpy/scripts/dx.py:3233]: https://confluence.internal.dnanexus.com/pages/viewpage.action?spaceKey=TTN&title=dx+run+workflow-xxxx+--instance-type-by-executable+--delay-workspace-destruction 
4.3.1 says "If "dx run --clone fromId"  is called with "--instance-type Y" input argument and there's a chance this --instance-type argument will be overridden by a combination of "--instance-type-by-executable X" and fromId.mergedSystemRequirementsByExecutable, print "Warning: --instance-type argument <Y> may get overridden by --instance-type-by-executable argument <X> and <fromId> mergedSystemRequirementsByExecutable value of <>".

The `if` statement may insert a warning event where there's no --clone (i.e. where there's both instance_type and instance_type_by_executable) - this is not what's speced

Comment (YuxinShi0423) [reply src/python/dxpy/scripts/dx.py]: Thanks! @emiloslavsky now it is fixed:
```
➜ dx run applet-GZv3JP801fX89p7Pkkzbb9Pq --instance-type mem1_sdd2_v2_x4 --clone job-GYBfXBQ00PB77fyv39FF719Q --instance-type-by-executable \{\"applet-GZv3JP801fX89p7Pkkzbb9Pq\":\{\"mai
n\":\"mem1_sdd2_v2_x2\"\}\} -y 

WARNING: --instance-type argument: mem1_sdd2_v2_x4 may get overridden by
job-GYBfXBQ00PB77fyv39FF719Q mergedSystemRequirementsByExecutable:
{"applet-GZv3JP801fX89...7Pkkzbb9Pq": {"main": "mem1_sdd2_x2"}} and runtime
--instance-type-by-executable argument:{"applet-GZv3JP801fX89p7Pkkzbb9Pq":
{"main": "mem1_sdd2_v2_x2"}}

➜ dx run applet-GZv3JP801fX89p7Pkkzbb9Pq --instance-type mem1_sdd2_v2_x4 --clone job-GYBfXBQ00PB77fyv39FF719Q -y 
WARNING: --instance-type argument: mem1_sdd2_v2_x4 may get overridden by
job-GYBfXBQ00PB77fyv39FF719Q mergedSystemRequirementsByExecutable:
{"applet-GZv3JP801fX89...7Pkkzbb9Pq": {"main": "mem1_sdd2_x2"}}
```

**Review by emiloslavsky — COMMENTED** [review]
Warning message looks good to me.  Deferring to engineering review for the rest.

**Comment by YuxinShi0423** [comment]
Confirmed by @evoropaeva that the fix works:

> I was using command the different way and everything is working fine now:
% my_venv/bin/dx run applet-GgQ92F00v2ZX1PbBzyKK5zj6 --instance-type-by-executable '{"applet-GgQ92F00v2ZX1PbBzyKK5zj6":{"main":"mem2_ssd1_v2_x32"}}'
fix can be deployed to stg

**Comment by evoropaeva** [comment]
@YuxinShi0423 could you please update this? I am waiting for your reply for nearly a  month already🥲

**Comment by YuxinShi0423** [comment]
@Evanj80 @ocbrewing Hi could you or anyone else help reviewing this fix? I already had it tested in jenkins (if needed I can rerun it and update the links under JIRA) and it should be good to go. Thanks @evoropaeva for the waiting!

File: src/python/test/test_dxclient.py [hunk 2]
```diff
@@ -3882,9 +3916,9 @@ def test_dx_run_clone_analysis(self):
         change_inst_type_analysis_id = run("dx run --clone " + analysis_id +
                                            " --instance-type mem2_hdd2_x2 --brief -y").strip()
         change_inst_type_by_exec_analysis_id = run("dx run --clone " + analysis_id +
-                                           " --instance-type-by-executable \'" + 
-                                           json.dumps({applet_id:{"*": {"instanceType": "mem2_ssd1_v2_x2"}}}) +
-                                           "\' --brief -y").strip()
+                                                   " --instance-type-by-executable \'" +
```
Comment (Evanj80) [inline src/python/test/test_dxclient.py:3919]: Maybe revert this tabbing?

**Review by Evanj80 — APPROVED** [review]

**Pull request merged**